File: app/src/main/java/com/example/myapplication/layout/AlertDialogActivity.java
package com.example.myapplication.layout;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.example.myapplication.R;

public class AlertDialogActivity extends AppCompatActivity {

    private Button button1;
    private Button button2;
    private Button button3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_alert_dialog);

        button1 = findViewById(R.id.button1);
        button2 = findViewById(R.id.button2);
        button3 = findViewById(R.id.button3);
        setOnClickListener();

    }

    private void setOnClickListener() {
        OnClick onClick = new OnClick();
        button1.setOnClickListener(onClick);
        button2.setOnClickListener(onClick);
        button3.setOnClickListener(onClick);
    }

    private class OnClick implements View.OnClickListener {

        @Override
        public void onClick(View v) {
            switch (v.getId()) {
                case R.id.button1:
                    AlertDialog.Builder builder1 = new AlertDialog.Builder(AlertDialogActivity.this);
                    builder1.setTitle("Question").setMessage("How do you feel").setPositiveButton("Great", new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            Toast.makeText(AlertDialogActivity.this, "Keep good feelings every day", Toast.LENGTH_SHORT).show();
                        }
                    }).setNegativeButton("Bad", new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            Toast.makeText(AlertDialogActivity.this, "forget the pain", Toast.LENGTH_SHORT).show();
                        }
                    }).setIcon(R.drawable.notification).show();
                    break;
                case R.id.button2:
                    AlertDialog.Builder builder2 = new AlertDialog.Builder(AlertDialogActivity.this);
                    final String[] array2 = {"male", "female"};
                    builder2.setTitle("Your sex").setSingleChoiceItems(array2, 0, new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            Toast.makeText(AlertDialogActivity.this, array2[which], Toast.LENGTH_SHORT).show();
                            dialog.dismiss();
                        }
                    }).setCancelable(false).show();
                    break;

                case R.id.button3:
                    AlertDialog.Builder builder3 = new AlertDialog.Builder(AlertDialogActivity.this);
                    final String[] array3 = {"basketball", "football", "tennis", "golf"};
                    boolean[] selected = {false, false, false, false};
                    builder3.setTitle("Your interests").setMultiChoiceItems(array3, selected, new DialogInterface.OnMultiChoiceClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which, boolean isChecked) {
                            Toast.makeText(AlertDialogActivity.this, array3[which], Toast.LENGTH_SHORT).show();
                        }
                    }).setPositiveButton("ok", new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {

                        }
                    }).setNegativeButton("cancel", new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {

                        }
                    }).show();
                    break;
            }
        }
    }

}

File: app/src/main/java/com/example/myapplication/layout/AlertDialogActivity.kt
package com.example.myapplication.layout

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.example.myapplication.R
import kotlinx.android.synthetic.main.activity_alert_dialog.buttonStyle1
import kotlinx.android.synthetic.main.activity_alert_dialog.buttonStyle2
import kotlinx.android.synthetic.main.activity_alert_dialog.buttonStyle3
import kotlinx.android.synthetic.main.activity_alert_dialog.buttonStyle4

class AlertDialogActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_alert_dialog)

        buttonStyle1.setOnClickListener(this)
        buttonStyle2.setOnClickListener(this)
        buttonStyle3.setOnClickListener(this)
        buttonStyle4.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.buttonStyle1 -> {
                val builder1 = AlertDialog.Builder(this@AlertDialogActivity)
                builder1.setTitle("Question").setMessage("How do you feel")
                    .setPositiveButton("Great") { dialog, which ->
                        Toast.makeText(
                            this@AlertDialogActivity,
                            "Keep good feelings every day",
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                    .setNegativeButton("Bad") { dialog, which ->
                        Toast.makeText(
                            this@AlertDialogActivity,
                            "forget the pain",
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                    .setIcon(R.drawable.notification).show()
                Log.d("AlertDialogActivity", "asynchronous message")
            }
            R.id.buttonStyle2 -> {
                val builder2 = AlertDialog.Builder(this@AlertDialogActivity)
                val array2 = arrayOf("male", "female")
                builder2.setTitle("Your sex").setSingleChoiceItems(array2, 0) { dialog, which ->
                    Toast.makeText(this@AlertDialogActivity, array2[which], Toast.LENGTH_SHORT)
                        .show()
                    dialog.dismiss()
                }.setCancelable(false).show()
            }
            R.id.buttonStyle3 -> {
                val builder3 = AlertDialog.Builder(this@AlertDialogActivity)
                val array3 = arrayOf("basketball", "football", "tennis", "golf")
                val selected = booleanArrayOf(false, false, false, false)
                builder3.setTitle("Your interests").setMultiChoiceItems(
                    array3,
                    selected
                ) { dialog, which, isChecked ->
                    Toast.makeText(
                        this@AlertDialogActivity,
                        array3[which],
                        Toast.LENGTH_SHORT
                    ).show()
                }
                    .setPositiveButton("ok") { dialog, which -> }
                    .setNegativeButton("cancel") { dialog, which -> }.show()
            }
            R.id.buttonStyle4 -> {
                startActivity(Intent(this, DialogActivity::class.java))
            }
        }
    }
}